packages/mdc-dialog/foundation.js

```diff
@@ -0,0 +1,206 @@
+      setFocusedElement: () => {},
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDCDialogFoundation.defaultAdapter, adapter));
+
+    this.lastFocusedElement_;
+    this.currentFocusedElIndex_ = -1;
+    this.inert_ = true;
+    this.isOpen_ = false;
+    this.componentClickHandler_ = () => this.cancel();
+    this.dialogClickHandler_ = (evt) => evt.stopPropagation();
+    this.acceptHandler_ = () => this.accept();
+    this.cancelHandler_ = () => this.cancel();
+    this.focusHandler_ = () => this.setFocus_();
+    this.documentKeydownHandler_ = (evt) => {
+      if (evt.key && evt.key === 'Escape' || evt.keyCode === 27) {
+        this.cancel();
+      }
+    };
+  }
+
+  init() {
+    const {OPEN} = MDCDialogFoundation.cssClasses;
+
+    if (this.adapter_.hasClass(OPEN)) {
+      this.isOpen_ = true;
+    } else {
+      this.lockTab_();
+      this.isOpen_ = false;
+    }
+  }
+
+  destroy() {}
+
+  getLastFocusTarget() {
+    return this.lastFocusedElement_;
+  }
+
+  open() {
+    this.lastFocusedElement_ = this.adapter_.getFocusedElement();
+    this.unlockTab_();
+    this.adapter_.registerAcceptHandler(this.acceptHandler_);
+    this.adapter_.registerCancelHandler(this.cancelHandler_);
+    this.adapter_.registerDocumentKeydownHandler(this.documentKeydownHandler_);
+    this.adapter_.registerDialogInteractionHandler('click', this.dialogClickHandler_);
+    this.adapter_.registerInteractionHandler('click', this.componentClickHandler_);
+    this.adapter_.registerFocusTrappingHandler(this.focusHandler_);
+    this.adapter_.setDefaultFocus();
+    this.disableScroll_();
+    this.adapter_.setBackgroundAttr('aria-hidden', true);
+    this.adapter_.setDialogAttr('aria-hidden', false);
+    this.adapter_.addClass(MDCDialogFoundation.cssClasses.OPEN);
+    this.isOpen_ = true;
+  }
+
+  close() {
+    this.lockTab_();
+    this.adapter_.deregisterAcceptHandler(this.acceptHandler_);
+    this.adapter_.deregisterCancelHandler(this.cancelHandler_);
+    this.adapter_.deregisterDialogInteractionHandler(this.dialogClickHandler_);
+    this.adapter_.deregisterDocumentKeydownHandler(this.documentKeydownHandler_);
+    this.adapter_.deregisterInteractionHandler(this.componentClickHandler_);
+    this.adapter_.deregisterFocusTrappingHandler(this.focusHandler_);
+    this.adapter_.removeClass(MDCDialogFoundation.cssClasses.OPEN);
+    this.enableScroll_();
+    this.adapter_.setBackgroundAttr('aria-hidden', false);
+    this.adapter_.setDialogAttr('aria-hidden', true);
+    this.isOpen_ = false;
+    this.adapter_.setFocusedElement(this.lastFocusedElement_);
+  }
+
+  accept() {
+    this.adapter_.acceptAction();
+    this.close();
+  }
+
+  cancel() {
+    this.adapter_.cancelAction();
+    this.close();
+  }
+
+  /**
+   *  Render all children of the dialog inert when it's closed.
+   */
+  lockTab_() {
+    if (this.inert_) {
+      return;
+    }
+
+    const elements = this.adapter_.getFocusableElements();
```

Comment:
This will all get extracted out once we build the focus trapping utility as well.